### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.ConcreteCategory.Bundled

### PR DESCRIPTION
**Pull request by TwoFX**

### CODE REVIEW

File: Mathlib/CategoryTheory/ConcreteCategory/Bundled.lean [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison, Johannes Hölzl, Reid Barton, Sean Leather
+-/
+import Std.Tactic.Lint.Frontend
+import Std.Tactic.Lint.Misc
+
+/-!
+# Bundled types
+
+`bundled c` provides a uniform structure for bundling a type equipped with a type class.
```
Comment (gebner) [inline Mathlib/CategoryTheory/ConcreteCategory/Bundled.lean:12]:
```suggestion
`Bundled c` provides a uniform structure for bundling a type equipped with a type class.
```

and below as well (`Category` too).

File: Mathlib/CategoryTheory/ConcreteCategory/Bundled.lean [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  /-- The underlying type of the bundled type -/
+  α : Type u
+  /-- The corresponding instance of the bundled type class -/
+  str : c α := by infer_instance
+
+namespace Bundled
+
+-- This is needed so that we can ask for an instance of `c α` below even though Lean doens't know
+-- that `c α` is a typeclass.
+set_option checkBinderAnnotations false
```
Comment (gebner) [inline Mathlib/CategoryTheory/ConcreteCategory/Bundled.lean:37]:
```suggestion
set_option checkBinderAnnotations false in
```

Then you don't need the `set_option false` below.

Comment (TwoFX) [reply Mathlib/CategoryTheory/ConcreteCategory/Bundled.lean]: Thanks, I didn't know about that.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**